test/functional/test_framework/test_node.py

```diff
@@ -97,14 +97,16 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
         ]
-        if use_valgrind:
+        if self.is_valgrind_present() and use_valgrind:
+            exit_on_first_error = self.should_exit_valgrind_on_first_error()
             default_suppressions_file = os.path.join(
                 os.path.dirname(os.path.realpath(__file__)),
                 "..", "..", "..", "contrib", "valgrind.supp")
             suppressions_file = os.getenv("VALGRIND_SUPPRESSIONS_FILE",
                                           default_suppressions_file)
             self.args = ["valgrind", "--suppressions={}".format(suppressions_file),
-                         "--gen-suppressions=all", "--exit-on-first-error=yes",
+                         "--gen-suppressions=all",
+                         "--exit-on-first-error=yes" if exit_on_first_error else "--exit-on-first-error=no",
                          "--error-exitcode=1", "--quiet"] + self.args
```

Comment:
nit: suggestion for readability, format the args array like in line 90 above:
```diff
-            self.args = ["valgrind", "--suppressions={}".format(suppressions_file),
-                         "--gen-suppressions=all",
-                         "--exit-on-first-error=yes" if exit_on_first_error else "--exit-on-first-error=no",
-                         "--error-exitcode=1", "--quiet"] + self.args
+            self.args = [
+                "valgrind",
+                "--suppressions={}".format(suppressions_file),
+                "--gen-suppressions=all",
+                "--exit-on-first-error=yes" if exit_on_first_error else "--exit-on-first-error=no",
+                "--error-exitcode=1",
+                "--quiet",
+            ] + self.args
```

Reply:
Yes! Thanks :)